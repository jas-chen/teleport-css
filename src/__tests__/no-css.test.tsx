import { expect, test } from 'vitest';
import { renderToStaticMarkup } from 'react-dom/server';
import fnv1a from '@sindresorhus/fnv1a';
import { create } from '..';
import pretty from 'pretty';

function hashFn(value: string) {
  return fnv1a(value, { size: 64 }).toString(36).slice(0, 8);
}

const { styled } = create({
  hashFn,
});

test('No css - null', () => {
  const Button = styled('button', () => null);

  expect(pretty(renderToStaticMarkup(<Button />))).toMatchSnapshot();

  expect(
    pretty(renderToStaticMarkup(<Button css={() => null} />)),
  ).toMatchSnapshot();
});

test('No css - string', () => {
  const Button = styled('button', () => 'hi');

  expect(pretty(renderToStaticMarkup(<Button />))).toMatchSnapshot();
});

test('No css - boolean', () => {
  const Button = styled('button', () => true);
  const Button2 = styled('button', () => false);

  expect(pretty(renderToStaticMarkup(<Button />))).toMatchSnapshot();
  expect(pretty(renderToStaticMarkup(<Button2 />))).toMatchSnapshot();
});

test('No css - nullish', () => {
  const Button = styled('button', () => null);
  const Button2 = styled('button', () => undefined);

  expect(pretty(renderToStaticMarkup(<Button />))).toMatchSnapshot();
  expect(pretty(renderToStaticMarkup(<Button2 />))).toMatchSnapshot();
});
