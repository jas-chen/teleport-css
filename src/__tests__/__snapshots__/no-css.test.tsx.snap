// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`No css - boolean 1`] = `"<button></button>"`;

exports[`No css - boolean 2`] = `"<button></button>"`;

exports[`No css - null 1`] = `"<button></button>"`;

exports[`No css - null 2`] = `"<button></button>"`;

exports[`No css - nullish 1`] = `"<button></button>"`;

exports[`No css - nullish 2`] = `"<button></button>"`;

exports[`No css - string 1`] = `"<button></button>"`;
